tests/unattended/unit/framework/bach.sh

```diff
@@ -0,0 +1,642 @@
+
+    bach_restore_stdin
+    @mockall "${bash_builtin_cmds[@]}" source .
+
+    eval "$(builtin export | while read -rs name; do
+        name="${name%%=*}"
+        name="${name##* }"
+        [[ "${name^^}" != BACH_* ]] || continue
+        builtin echo "unset '$name' || builtin true"
+    done)"
+    builtin export LANG=C TERM=vt100
+    unset name util_path
+}
+
+function @real() {
+    declare name="$1" real_cmd
+    if [[ "$name" == */* ]]; then
+        @echo "$@"
+        return
+    fi
+    real_cmd="$(bach-real-path "$1" 7>&1)"
+    if [[ -z "${real_cmd}" ]]; then
+        real_cmd="${name}_not_found"
+    fi
+    declare -a cmd=("${real_cmd}" "${@:2}")
+    @debug "[REAL-CMD]" "${cmd[@]}"
+    "${cmd[@]}"
+}
+export -f @real
+
+function bach-get-all-functions() {
+    declare -F
+}
+export -f bach-get-all-functions
+
+function bach--skip-the-test() {
+    declare test="$1" test_filter
+    while read -d, test_filter; do
+        [[ -n "$test_filter" ]] || continue
+        [[ "$test" == $test_filter ]] && return 0
+        [[ "$test" == test-$test_filter ]] && return 0
+    done <<< "${BACH_TESTS:-},"
+}
+export -f bach--skip-the-test
+
+function bach-run-tests--get-all-tests() {
+    bach-get-all-functions | @sort -R | while read -r _ _ name; do
+        [[ "$name" == test?* ]] || continue
+        [[ "$name" == *-assert ]] && continue
+        bach--skip-the-test "$name" || continue
+        builtin printf "%s\n" "$name"
+    done
+}
+
+for donotpanic in donotpanic dontpanic do-not-panic dont-panic do_not_panic dont_panic; do
+    eval "function @${donotpanic}() { builtin printf '\n%s\n  line number: %s\n  script stack: %s\n\n' 'DO NOT PANIC!' \"\${BASH_LINENO}\" \"\${BASH_SOURCE[*]}\"; builtin exit 1; } >&2; export -f @${donotpanic};"
+done
+
+function bach--is-function() {
+    [[ "$(@type -t "$1")" == function ]]
+}
+export -f bach--is-function
+
+declare -gr __bach_run_test__ignore_prefix="## BACH:"
+function @comment() {
+    @out "${__bach_run_test__ignore_prefix}" "$@"
+}
+export -f @comment
+
+function bach-run-tests() {
+    set -euo pipefail
+
+    bach_initialize
+
+    for donotpanic in donotpanic dontpanic do-not-panic dont-panic do_not_panic dont_panic; do
+        eval "function @${donotpanic}() { builtin true; }; export -f @${donotpanic}"
+    done
+
+    function command() {
+        if [[ "$1" != -* ]] && bach--is-function "$1"; then
+            "$@"
+        else
+            mockfunc="$(@generate_mock_function_name command "$@")"
+            if bach--is-function "${mockfunc}"; then
+                @debug "[BC-func]" "${mockfunc}" "$@"
+                "${mockfunc}" "$@"
+            else
+                command_not_found_handle command "$@"
+            fi
+        fi
+    }
+    export -f command
+
+    function xargs() {
+        declare param
+        declare -a xargs_opts
+        while param="${1:-}"; [[ -n "${param:-}" ]]; do
+            shift || true
+            if [[ "$param" == "--" ]]; then
```

Comment:
Same

Reply:
`bach.sh` is the external framework. We will not modify it.